src/include/sof/audio/audio_stream.h

```diff
@@ -338,6 +384,21 @@ audio_stream_get_free_frames(const struct audio_stream *stream)
 		audio_stream_frame_bytes(stream);
 }
 
+/**
+ * Calculates free space in frames aligned. the stream->frame_align_shift-th
+ * power of 2 is the "frame byte" which is not less than stream->frame_align
+ * * audio_stream_frame_bytes(stream), using right shift instead of division,
+ * handling underrun_permitted behaviour
+ * @param stream Stream pointer
+ * @return amount of space free in frames
+ */
+static inline uint32_t
+audio_stream_get_free_frames_aligned(const struct audio_stream *stream)
```

Comment:
These two nearly identical functions have a very long doxygen which makes them look like an API but I feel like they're meant to be used only by `audio_stream_avail_frames_aligned()`, correct? If correct then move their code directly into `audio_stream_avail_frames_aligned()` and get rid of these two functions.